bulk-model-sync-gradle/src/test/kotlin/org/modelix/model/sync/bulk/gradle/MpsRunnerConfigurationTest.kt

```diff
@@ -0,0 +1,95 @@
+package org.modelix.model.sync.bulk.gradle
+
+import io.kotest.matchers.collections.shouldContainExactlyInAnyOrder
+import org.modelix.model.sync.bulk.gradle.config.LocalTarget
+import org.modelix.model.sync.bulk.gradle.config.ServerSource
+import org.modelix.model.sync.bulk.gradle.config.SyncDirection
+import java.io.File
+import kotlin.test.Test
+
+class MpsRunnerConfigurationTest {
+
+    private val jsonDir = File("/jsonDir")
+    private val classPathElements = emptySet<File>()
+
+    @Test
+    fun `build configuration for local target with a server source`() {
+        val serverSource = ServerSource()
+        val localTarget = LocalTarget(repositoryDir = File("/repositoryDir"))
+        val syncDirection = SyncDirection("syncDirection", serverSource, localTarget)
+
+        val config = buildMpsRunConfigurationForLocalTarget(syncDirection, classPathElements, jsonDir)
+        val jvmArgs = config.jvmArgs
+
+        val expectedJvmArgs = listOf(
+            "-Dmodelix.mps.model.sync.bulk.input.path=/jsonDir",
+            "-Dmodelix.mps.model.sync.bulk.input.modules=",
+            "-Dmodelix.mps.model.sync.bulk.input.modules.prefixes=",
+            "-Dmodelix.mps.model.sync.bulk.repo.path=/repositoryDir",
+            "-Dmodelix.mps.model.sync.bulk.input.continueOnError=false",
+            "-Xmx2g",
+        )
+
+        jvmArgs shouldContainExactlyInAnyOrder expectedJvmArgs
+    }
+
+    @Test
+    fun `build configuration for local target with a server source with a base revision and branch name`() {
+        val serverSource = ServerSource(
+            url = "aUrl",
+            repositoryId = "aRepositoryId",
+            branchName = "aBranchName",
+            baseRevision = "aBaseRevision",
```

Comment:
We also need similar test cases for `baseRevision=null`, as it is also valid to provide a revision or pair of repoId and branchName without a baseRevision.

These cases are broken right now with this PRs changes.
The fact that this isn't covered by existing tests right now emphasizes our need to be able to test multiple gradle configurations.

Reply:
> as it is also valid to provide a revision or pair of repoId and branchName without a baseRevision.

If a `baseRevision` is not provided, the `repositoryId` and `branchName` are not passed to the `MPSBulkSynchronizer`. The data is downloaded in a separate MPS task.

So that case should not be broken.

> The fact that this isn't covered by existing tests right now emphasizes our need to be able to test multiple gradle configurations.

Definitely. But for now, this is out of scope for me.

Reply:
> If a baseRevision is not provided, the repositoryId and branchName are not passed to the MPSBulkSynchronizer.

Ah yes, you are right it is only relevant for the `ExportFromModelServerTask`.

> But for now, this is out of scope for me.

Agreed.

Reply:
> If a baseRevision is not provided, the repositoryId and branchName are not passed to the MPSBulkSynchronizer.

I update the test case `build configuration for local target with a server source without a base revision` to test this, just to be sure.